### PR TITLE
Feat/anonymized item exchange bruno tests

### LINKED ISSUE #1081: [Story] Bruno tests for anonymized item stock exchange
**As** PURIS admin,
**I want** bruno integration tests for anonymized stocks,
**so that** I can test the full exchange of anonymized data.

## Hints / Details

- the tests should be added as a new folder to the test section `Test_02-EDC`
- the test should include the following sections:
  - contract setup
  - subdmodel descriptor retrieval (including DTR negotiation)
  - retrieval of submodel data (including Negotiation for submodel Asset with the EDC)

### Outcome

- anonymized stock data is exchanged successfully

### Acceptance Criteria

- variables like the semantic id are added as environment variables where appropriate
- key tests verify the structure of returned data models
- all tests are tagged with the new tag `exchange-anonymized-data`

## Out of Scope

- edge cases (the focus is on the successful exchange scenario)
- test suite decoumentation is to be handled separately due to the large volume of changes

### PR DESCRIPTION
**Pull request by OlgaIvkovic**
## Description
- added Bruno tests for anonymized item stock
- the tests consist of 3 phases:
    - contract creation
    - DTR negotiation and retrieval of submodel descriptors
    - submodel asset negotiation and retrieval of submodel data

Resolves #1081 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files ([TRG 7.02](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02)
- [x] Documentation Notice are present on all affected files ([TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07))
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).
- [x] **Changelog** updated (`changelog.md`) with PR reference and brief summary.
- [x] **Frontend** version bumped, if needed (`frontend/package.json`, `frontend/package-lock.json`)
- [x] **Backend** version bumped, if needed (`backend/pom.xml`)
- [x] **Open API** specification updated, if controllers have been changed (use python script `scripts/generate_openapi_yaml.py` with running customer backend)

### CODE REVIEW

File: local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/01-Contract setup/02-ItemStock/03-Create Contract Definitiion.bru [hunk 1]
```diff
@@ -0,0 +1,38 @@
+meta {
+  name: 03-Create Contract Definitiion
+  type: http
+  seq: 1
+  tags: [
+    share-enablement-services
```
Comment (ReneSchroederLJ) [inline local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/01-Contract setup/02-ItemStock/03-Create Contract Definitiion.bru]:
```suggestion
   exchange-anonymized-data
```
Please also apply this to the contract definition in delivery and add the tags to request 01 and 02

File: local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-ItemStock/01-Retrieve Submodel Descriptor from DTR/01-Query Catalog for DTR.bru [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        "odrl:obligation": [],
+        "odrl:target": {
+            "@id": assetId
+        }
+    }
+
+    bru.setVar("DTR_CATALOG_OFFER", JSON.stringify(offer))
+    bru.setVar("DTT_CATALOG_OFFER_ID", offerId)
+  } else {
+    bru.runner.setNextRequest("04-Initiate Transfer")
```
Comment (ReneSchroederLJ) [inline local/bruno/puris-integration-test/Test_02-EDC/Exchange anonymized submodel data/02-Contract Negotiation/02-ItemStock/01-Retrieve Submodel Descriptor from DTR/01-Query Catalog for DTR.bru]: this functionality seems to be causing a loop, because it mistakenly jumps back to the request within the delivery subsection. Please remove the if and else clause for now and just always run the logic.

**Review by ReneSchroederLJ — CHANGES_REQUESTED** [review]
Thank you for your contribution. Looks good for the most part. There's one small script problem causing a loop and a few missing tags.

**Comment by ReneSchroederLJ** [comment]
Thank you for your contribution. It has been decided to reuse the tests from deliveries for all anonymized submodels. This was implemented in #1130. This PR is no longer necessary.
